allow puppetlabs/firewall 8.x

jhoblitt opened the pull request.

jhoblitt committed:
allow puppetlabs/firewall 8.x

jhoblitt commented:
I believe the CI failure(s) are unrelated to the PR.
```
    Problem: package perfsonar-toolkit-5.1.2-1.el8.noarch from perfSONAR requires perfsonar-core, but none of the providers can be installed
    - package perfsonar-core-5.1.2-1.el8.noarch from perfSONAR requires perfsonar-testpoint, but none of the providers can be installed
    - package perfsonar-testpoint-5.1.2-1.el8.noarch from perfSONAR requires httpd-wsgi-socket, but none of the providers can be installed
    - cannot install the best candidate for the job
    - nothing provides mod_wsgi needed by httpd-wsgi-socket-1.0-1.el8.4261.77c636c73c9f8535e20ed171f070cf5057242ea4.noarch from perfSONAR
    - nothing provides mod_wsgi needed by httpd-wsgi-socket-1.0-1.el8.noarch from perfSONAR
```

treydock commented:
The unit tests need updates to account for different firewall property being used.

treydock commented:
Ah, update `.fixtures.yml` to not pin to 6.0.0, I think that's why the unit tests are failing.

jhoblitt committed:
modernize .fixtures.yml and remove version pins

jhoblitt commented:
I think the provider metaparam went away in >=  7.0.0: https://github.com/puppetlabs/puppetlabs-firewall/blob/main/CHANGELOG.md#v700---2023-09-13

jhoblitt committed:
rm empty data/common.yaml

Resolves this warning:

    Warning: /etc/puppetlabs/code/modules/perfsonar/data/common.yaml: file does not contain a valid yaml hash

jhoblitt commented:
EL8 is still broken:
```
    Problem: package perfsonar-toolkit-5.1.2-1.el8.noarch from perfSONAR requires perfsonar-core, but none of the providers can be installed
    - package perfsonar-core-5.1.2-1.el8.noarch from perfSONAR requires perfsonar-testpoint, but none of the providers can be installed
    - package perfsonar-testpoint-5.1.2-1.el8.noarch from perfSONAR requires httpd-wsgi-socket, but none of the providers can be installed
    - cannot install the best candidate for the job
    - nothing provides mod_wsgi needed by httpd-wsgi-socket-1.0-1.el8.4261.77c636c73c9f8535e20ed171f070cf5057242ea4.noarch from perfSONAR
    - nothing provides mod_wsgi needed by httpd-wsgi-socket-1.0-1.el8.noarch from perfSONAR
```

I don't need to support pS on EL8.  I don't suppose this module could just drop support for it?  Otherwise, the acceptance tests probably need to setup additional yum repos for EL8.

treydock commented:
Looks like PerfSONAR repos are broken for EL8.  My EL8 perfSonar server has the mod_wsgi RPM from when it was provisioned using PerfSONAR repos but now the RPM is missing.

treydock commented:
I'll merge even with EL8 failure.  I'm working on some stdlib 9.x support I need to release first as I'm not quite ready to jump to firewall >= 7.0.  Doing stdlib 9.x upgrade first.

treydock commented on manifests/firewall.pp:
```diff
@@ -56,7 +56,7 @@
       chain    => 'INPUT',
       jump     => 'perfSONAR',
-      provider => 'ip6tables',
+      protocol => 'ip6tables',
```
```suggestion
      protocol => 'IPv6',
```

treydock replied:
Based on firewall module migration to 7.0.0.  Says `ip6tables` will work but docs now use `IPv6`

treydock commented on manifests/firewall.pp:
```diff
@@ -71,8 +71,8 @@
-          action   => 'accept',
-          provider => 'ip6tables',
+          jump     => 'accept',
+          protocol => 'ip6tables',
```
```suggestion
          protocol => 'IPv6',
```

treydock commented:
Updating in #29.  I'm still behind on some updates and using firewall 6.x so made this module support both old and new hopefully.
